test/tests/utilities.js

```diff
@@ -272,5 +272,89 @@ describe("Zotero.Utilities", function() {
 			
 			assert.isUndefined(cslJSON.PMID, 'field labels are case-sensitive');
 		});
+		it("should parse particles in creator names", function() {
+			let creators = [
+				{
+					// No particles
+					firstName: 'John',
+					lastName: 'Smith',
+					creatorType: 'author',
+					expect: {
+						given: 'John',
+						family: 'Smith'
+					}
+				},
+				{
+					// dropping and non-dropping
+					firstName: 'Jean de',
+					lastName: 'la Fontaine',
+					creatorType: 'author',
+					expect: {
+						given: 'Jean',
+						"dropping-particle": 'de',
+						"non-dropping-particle": 'la',
+						family: 'Fontaine'
+					}
+				},
+				{
+					// only non-dropping
+					firstName: 'Vincent',
+					lastName: 'van Gogh',
+					creatorType: 'author',
+					expect: {
+						given: 'Vincent',
+						"non-dropping-particle": 'van',
+						family: 'Gogh'
+					}
+				},
+				{
+					// only dropping
+					firstName: 'Alexander von',
+					lastName: 'Humboldt',
+					creatorType: 'author',
+					expect: {
+						given: 'Alexander',
+						"dropping-particle": 'von',
+						family: 'Humboldt'
+					}
+				},
+				{
+					// institutional author
+					lastName: 'Jean de la Fontaine',
+					creatorType: 'author',
+					fieldMode: 1,
+					expect: {
+						literal: 'Jean de la Fontaine'
+					}
+				},
+				{
+					// protected last name
+					firstName: 'Jean de',
+					lastName: '"la Fontaine"',
+					creatorType: 'author',
+					expect: {
+						given: 'Jean de',
+						family: 'la Fontaine'
+					}
```

Comment:
Sorry about the update to the name parser; I addressed a user request and fixed a bug. The name parser should probably be spun off to a separate project, but for now I'll just be careful not to touch it further without running changes past everyone affected.

About the dropping-particle no-parse, I noticed that in the code, but didn't look into it because the processor tests were passing. I think that there are currently no name formats in which the given name is not followed immediately by the dropping-particle, which would explain it. It *should* parse out, though, to be correct.

I don't have time to rework the code at the moment - I have my hands full with the Zotero 5.0 migration of Juris-M - but pull requests against the processor code are welcome.